test-helpers.bash

```diff
@@ -153,6 +153,43 @@ function wait_for_container_output() {
     done
 }
 
+function wait_for_container_output_matching_file() {
+    local MAX_ATTEMPTS=$1
+    local CONTAINER_NAME=$2
+    local EXPECTED_OUTPUT_FILE=$3
+
+    shift
+    local ATTEMPTS=0
+
+    # This function may be run outside of BATS, so ensure `fail` has a definition
+    if [[ $(type -t fail) != function ]]; then
+        function fail() {
+            local message=$1
+            echo "FAIL: $message"
+            exit 1
+        }
+    fi
+
+    if ! "$CONTAINER_RUNTIME" logs "$CONTAINER_NAME" ; then
+        fail "unable to get logs for container: $CONTAINER_NAME"
+    fi
+
+    # Pipe both "docker logs" and expected_output through tr replacing newlines with non-printable byte. This
+    # allows grep to match a multline string against docker logs output
+    # Note for failing containers, logs go to stderr
+    until ("$CONTAINER_RUNTIME" logs "$CONTAINER_NAME" || :) 2>&1 | tr '\n' '\1' | grep -qF "$(tr '\n' '\1' < $EXPECTED_OUTPUT_FILE)" ; do
+        # Prevent an infinite loop - at 0.2 seconds per go this is 10 minutes
+        if [ $ATTEMPTS -gt "3000" ]; then
+            fail "wait_for_container_output_matching_file ultimate max exceeded: \"$(cat $EXPECTED_OUTPUT_FILE)\" ($*)"
+        fi
+        if [ $ATTEMPTS -gt "$MAX_ATTEMPTS" ]; then
+            fail "wait_for_container_output_matching_file unable to find output: \"$(cat $EXPECTED_OUTPUT_FILE)\" ($*)"
+        fi
+        ATTEMPTS=$((ATTEMPTS+1))
+        sleep 0.2
```

Comment:
Can we default it but allow a configurable input/env var then as well? I'm thinking some usage might take a while and others might be fast.